vktrace: 32 bit build fix

johnzupin opened the pull request:
Problem was discovered in GH issue #371

vktrace/CMakeList.txt: fix conditional on 32bit build check 

vktrace/vktrace_layer/CMakeLists.txt: change cmake function call to
find_library

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/LunarG/VulkanTools?pullRequest=374) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/LunarG/VulkanTools?pullRequest=374) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/LunarG/VulkanTools?pullRequest=374) it.</sub>

johnzupin commented:
Karl I was told to assign this to you. If @davidlunarg or @joey-lunarg approve this I can probably merge it.

johnzupin commented:
This might be poorly named PR, the second part i changed
```
-       pkg_search_module(XCB-KEYSYMS REQUIRED xcb-keysyms)
+       find_library(XCB-KEYSYMS REQUIRED xcb-keysyms)
```
in vktrace/vktrace_layer/CMakeLists.txt.

It initially couldn't find the XCB-KEYSYMS library until I changed it to find_library.

karl-lunarg commented on vktrace/vktrace_layer/CMakeLists.txt:
```diff
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 2.8)
    if (BUILD_WSI_XCB_SUPPORT)
        set(ENV{VULKAN_WSI} "Xcb")
-       pkg_search_module(XCB-KEYSYMS REQUIRED xcb-keysyms)
+       find_library(XCB-KEYSYMS REQUIRED xcb-keysyms)
```
find_package  is probably preferred here.   find_package will set vars that help CMake locate headers and libraries, whereas find_library will set vars that help CMake locate just the libraries.

I'd suggest trying find_package and use it if it works.

johnzupin commented:
@karl-lunarg  I did something a bit different, I put the job of finding libxcb-keysyms in FindXCB.cmake. FindXCB.cmake looks for components and since libxcb-keysyms1-dev is a required package in the docs I was thinking that's were it should be. 

I did have to change a line in FindXCB.cmake to look for "xcb_keysyms.h" instead of "keysyms.h". That seems to be what the header file is called at least on Ubuntu.

Let me know if this is wrong.

karl-lunarg commented:
I'd be afraid of the cmake/findXCB.cmake change not working on other distros.

I just tried it with the change in the conditional and changing pkg_search_module to find_library, and it worked.  Plus the OP said that find_library worked.  So I'd favor the find_library change.

johnzupin commented:
@karl-lunarg I made these changes and the tests pass. Let me know if you approve.
